nntrainer/tensor/cl_operations/testing_rotarty_emb.cpp

```diff
@@ -0,0 +1,181 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2024 Yash Singh <yash.singh@samsung.com>
+ *
+ * @file	testing_rotary_emb.cpp
+ * @date	28 August 2024
+ * @brief	Rotary Embedding CPU code
+ * @see		https://github.com/nnstreamer/nntrainer
+ * @author	Yash Singh <yash.singh@samsung.com>
+ * @bug		No known bugs except for NYI items
+ *
+ */
+
+#include "tensor.h"
+#include <string>
+
+/**
+ * @brief     compute frequency for rotary embedding
+ * @param[in] dim hidden dim size
+ * @param[in] seq_len sequency length
+ * @param[out] freqs_cos cosine of the frequencies
+ * @param[out] freqs_sin sine of the frequencies
+ * @param[out] freqs base frequencies array to be used in computation of cos and
+ * sin values for each position in sequence
+ * @param[in] theta rotary angle
+ */
+void precompute_freqs(int dim, unsigned int seq_len,
```

Comment:
This function shares the same implementation with precompute_freqs() in attention_kernel_interface.cpp. could you explain why you redefined it here?

Reply:
So Initially, testing_rotary_emb.cpp was to test the cpu code before writing the kernel. No specific reason for redefining it again. We can remove it as well, I didn't as initially in gpu kernel the paramters were different as compared to the cpu one. So thats why kept both. It was just for testing purposes.

Reply:
makes sense! I think a comment or a brief note that this is a testing code for CPU results would be awesome :)